test/test.js

```diff
@@ -1,5 +1,52 @@
+const path = require("path")
 const test = require("tape").test
+const FlyNotify = require("../")
+const defaults = {
+  title: "Fly",
+  message: "Task completed.",
+  icon: path.resolve(__dirname, "..", "fly-logo.png"),
+}
+const plugin = new FlyNotify()
 
-test("fly-notify", function (t) {
- t.end()
+// Adding this to make sure tests pass since we're not checking that the `node-
+// notifier` functionality works.
+plugin.error = function() {}
+plugin.debug = function() {}
+
+test("Testing `fly-notify`", function (t) {
+  t.plan(2)
+  t.ok(plugin.notify, "Fly-Notify `notify` method exists ~~%")
+  t.equal(typeof plugin.notify, "function", "Fly-Notify `notify` method is a function ~~%")
+  t.end()
+})
+
+test("Testing bad options", function(t) {
```

Comment:
Looks basically good to me. @rogeruiz If by framework you mean guidelines, then we are cool. Basically, Fly tests whether filters work, so no fly plugin should run fly and check whether the filter works, instead they should check whether the transform function inside the filter produces the correct output. Most plugins are thin wrappers around any given transform library so these tests should be trivial. I can imagine they will be more complicated for plugins other than transforms/writers.